### ISSUE TITLE
bug: super(arg) can't jump

### ISSUE THREAD

**Issue by SolaWing** [castwide/solargraph#280]
```ruby
class A
  def initialize(*args); end
end

class B < A
  def initialize
    super # can jump
    super 2 # can't jump
    super(1) # can't jump too
  end
end
```

**Comment by castwide** [comment]
Confirmed. This should be a simple fix. The NodeChainer doesn't recognize `super` with arguments because the parser assigns it a different node type than `super` without arguments.

**Comment by castwide** [comment]
Fixed in the v0.39 branch. There are still some changes I want to make around `super` argument validation in the type checker, but jumping to definition should work as expected.
